app/controllers/users/edit_password_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Users
+  class EditPasswordController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def edit
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user)
+    end
+
+    def update
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user)
+
+      result = @update_user_password_form.submit(user_params)
+
+      analytics.track_event(:password_change, result)
+
+      if result[:success?]
+        handle_success
+      else
+        render :edit
```

Comment:
Yes, on the web page. The controller doesn't have to do anything special. The errors come from the Form Object and are displayed inline in the form via Simple Form, not as a flash message. This is how all our other forms work. See the screenshots that I will post shortly in the PR.